include/seqan3/range/views/kmer_hash.hpp

```diff
@@ -251,14 +251,17 @@ class kmer_hash_view<urng_t>::shape_iterator
     *
     * Linear in size of shape.
     */
-    shape_iterator(it_t it_start, shape s_) :
-        shape_{s_}, text_left{it_start}, text_right{it_start}
+    shape_iterator(it_t it_start,  sentinel_t it_end, shape s_) :
+        shape_{s_}, text_left{it_start}, text_right{std::ranges::next(it_start, s_.size(), it_end)}
     {
         assert(std::ranges::size(shape_) > 0);
 
-        roll_factor = pow(sigma, static_cast<size_t>(std::ranges::size(shape_) - 1));
+        if (shape_.size() <= std::distance(text_left, text_right))
```

Comment:
```suggestion
        if (shape_.size() <= std::ranges::distance(text_left, text_right))
```
Beacuse it is the constrained version.